torch/distributed/_composable/contract.py

```diff
@@ -7,18 +7,12 @@
 from torch.distributed._composable_state import _State
 
 
-# use state_slot as key for module.__dict__ to avoid coliding with other
-# properties.
-# TODO: since all composable distributed features can share the same slot.
-class _StateKey(str):
-    # Make _StateKey as str to satisfy the assumption that object.__dict__.keys()
-    # are strings.
-    def __new__(cls, string="__composable_api_state_key"):
```

Comment:
@fegin @rohan-varma Please tag whoever may know. I was trying to figure out why this test fails, and it turns out that we are relying on the behavior that when a statekey gets deepcopied, it takes on a new uuid value.

Specifically, if the original STATE_KEY is plugged into a module which we deepcopy, we expect STATE_KEY+"_"+uuid() to be on the copied module.  

I'm a little lost as to what this whole thing is in service of, and why we want this copy behavior

Reply:
I don't know the initial intention but with `STATE_KEY` being an object, one can ensure that the existing states will be invalidated after the deepcopy. So users can apply `replicate` to the model again. @mrshenli can you share the original intention of making `STATE_KEY` being a new object?

Reply:
> @mrshenli can you share the original intention of making STATE_KEY being a new object?

Hey @fegin @wconstab, the original intention was to avoid collision with user-defined attr name strings, as `f"{string}_{str(uuid.uuid4())}"` in theory does not prevent collision.

Curious, why do we need to deepcopy _StateKey? I assume this should be a trainer-specified config, and should be decoupled from model state dicts?

Reply:
@mrshenli  There is an unittest that do `copied_model = deepcopy.copy(model)` after `replicate(model)` is called. `_StateKey` will be copied because the deepcopy of the model. It seems to me that the unittest intends to test the deepcopy of a replicated model.

Reply:
> existing states will be invalidated after the deepcopy. So users can apply replicate to the model again

could we take a different direction here, if this is the main issue?  can we just provide an api to 'clear' the composable tags from a copied model?  It seems to me like the 'expected' behavior for a deepcopy would be that you'd get an identical copy (e.g. still has annotations on it, if the original did, and they are still valid).  If some users want to clear them and start over, we could make that explicit.  Then we could also simplify the statekey.

Reply:
https://github.com/pytorch/pytorch/pull/105503 change the test and this PR should be unblocked.